### ISSUE TITLE
Project specific cd_user

### ISSUE THREAD

**Issue by michaelsauter** [opendevstack/ods-project-quickstarters#134]
We had some issues with a global `cd_user` (e.g. if password is changed, there are lots of places that need to be adjusted). This would be easier if each project had its own `cd_user`. 

Just raising this question / issue here - not sure how that would work and what implications this has.

FYI @clemensutschig @gerardcl @stitakis @rattermeyer

**Comment by gerardcl** [comment]
I think we already tackled options at the very beginning but there was no chance to create/promote securely rights to a newly created cd_user_<project> from main cd_user. 

We could tackle this again since there are new changes/features on current oc version.

What we could also think is about how to deal with password expiration/change policies, so to have an automated "cron" job that deals with that, globally, for system users.

The main point here, I would say, is how to dynamically manage the user creation and  role bindings, if required.

**Comment by clemensutschig** [comment]
We could do - prov App could generate the User and pass the credentials down to rundeck to create the secret. Thoughts?

**Comment by michaelsauter** [comment]
Yes that sounds good to me! @rattermeyer @stitakis @tjaeschke Other thoughts on this?

**Comment by tjaeschke** [comment]
> Yes that sounds good to me! @rattermeyer @stitakis @tjaeschke Other thoughts on this?

Sounds good to me at first glance

**Comment by clemensutschig** [comment]
I move this to provision app - as leading change source ...

impacted pieces:
- ods-project-quickstarters/rundeck:
 https://github.com/opendevstack/ods-project-quickstarters/blob/master/rundeck-jobs/openshift/create-projects.yaml#L32
- ods-project-quickstarters/create.projects.sh:
https://github.com/opendevstack/ods-project-quickstarters/blob/master/ocp-templates/scripts/create-projects.sh#L105
- ods-project-quickstarters/jenkins-master-template:
https://github.com/opendevstack/ods-project-quickstarters/blob/master/ocp-templates/templates/cd-jenkins-master.yml#L13-L22
- jenkins-shared-library:
https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/OdsContext.groovy#L301
https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/OdsContext.groovy#L359
- ods-provision-app
https://github.com/opendevstack/ods-provisioning-app/blob/master/src/main/java/org/opendevstack/provision/services/RundeckAdapter.java#L162

so this is a huge one :) @michaelsauter  fyi
